src/action/finder/hash.rs

```diff
@@ -0,0 +1,190 @@
+use crate::action::finder::request::HashActionOptions;
+use crate::fs::Entry;
+use crypto::digest::Digest;
+use log::warn;
+use rrg_proto::file_finder_hash_action_options::OversizedFilePolicy;
+use rrg_proto::Hash as HashEntry;
+use std::cmp::min;
+use std::fs::File;
+use std::io::Read;
+
+/// Hashes data writen to it using SHA-1, SHA-256 and MD5 algorithms.
+struct Hasher {
+    sha1: crypto::sha1::Sha1,
+    sha256: crypto::sha2::Sha256,
+    md5: crypto::md5::Md5,
+    num_bytes: usize,
+}
+
+impl Hasher {
+    pub fn new() -> Hasher {
+        Hasher {
+            sha1: crypto::sha1::Sha1::new(),
+            sha256: crypto::sha2::Sha256::new(),
+            md5: crypto::md5::Md5::new(),
+            num_bytes: 0,
+        }
+    }
+}
+
+impl std::io::Write for Hasher {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        self.sha1.input(buf);
+        self.sha256.input(buf);
+        self.md5.input(buf);
+        self.num_bytes = self.num_bytes + buf.len();
+
+        Ok(buf.len())
+    }
+
+    fn flush(&mut self) -> std::io::Result<()> {
+        Ok(())
+    }
+}
+
+pub fn hash(entry: &Entry, config: &HashActionOptions) -> Option<HashEntry> {
```

Comment:
Public functions should be documented (private ones preferably as well).

Reply:
Added docs.